### PR TITLE
feat: Improving Tax Modal and Including company in Cards

### LINKED ISSUE #2798: Event Wizard Step 1: Add Tax Info - Increase height and show all info
The tax info in the wizard step 1 ..
a) .. needs a scrollable field. Please increase the height of the field to minimize the need for scrolling.
b) Show all the tax info including company details once the field has been saved.

Screenshot from 2019-04-25 00-19-38

Screenshot from 2019-04-25 00-22-03

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #2798 

#### Screenshot:

image

#### Changes proposed in this pull request:

- Added company information in Tax Cards
- Added Validation for tax rate
- Removed scrolling as requested

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by kushthedude** [comment]
@shreyanshdwivedi @uds5501 Need a small help, I was trying to use
```
<div class="{{if tax.shouldSendInvoice 'y-scrollable scrollable-' 'modal'}} content">
```
But this thing is not working and creating an extra overflow, Any method to tackle it

File: app/templates/components/modals/tax-info-modal.hbs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <div class="header">
   {{t 'Add tax information'}}
 </div>
-<div class="y-scrollable scrollable-modal content">
+<div class="modal content">
```
Comment (kushthedude) [inline app/templates/components/modals/tax-info-modal.hbs]: @shreyanshdwivedi @uds5501 Need a small help, I was trying to use
```
<div class="{{if tax.shouldSendInvoice 'y-scrollable scrollable-' 'modal'}} content">
```
But this thing is not working and creating an extra overflow, Any method to tackle it

Comment (uds5501) [reply app/templates/components/modals/tax-info-modal.hbs]: @kushthedude I am curious as to how did you increased the size of the modal.

Comment (kushthedude) [reply app/templates/components/modals/tax-info-modal.hbs]: @uds5501 I didn't increase the size, I just switched off scrolling but its not valid fix , thats why asking if the code i have written above can be applied with some changes ?

Comment (uds5501) [reply app/templates/components/modals/tax-info-modal.hbs]: @kushthedude should this code be using `data.event.tax.shouldSendInvoice` ?

Comment (kushthedude) [reply app/templates/components/modals/tax-info-modal.hbs]: Since i am already under `Tax modal` , So data.event is not needed here .

Comment (kushthedude) [reply app/templates/components/modals/tax-info-modal.hbs]: @shreyanshdwivedi @mrsaicharan1 any idea ?

**Comment by kushthedude** [comment]
@CosmicCoder96 I checked the `Semantic UI ` and they provide no method to change the Height for modals , Can you suggest something ?

**Comment by niranjan94** [comment]
> @CosmicCoder96 I checked the Semantic UI and they provide no method to change the Height for modals , Can you suggest something ?

@kushthedude you can always use plain old css 😄

**Comment by kushthedude** [comment]
@niranjan94 Sure .

**Comment by niranjan94** [comment]
> I tried adding importing a sass file for modal but was not working , then I tried adding custom css to the `Div Container` of `Tax Modal` and it was not working too . As given in `UI Documentation` the modal have self adjusting providing no attribute to alter it .
> Any other way ?

@kushthedude increase the height of the `content` div via css. That should work

**Comment by kushthedude** [comment]
@niranjan94 Custom CSS are worsening it , Current changes seems much better than that . Please Review

**Comment by niranjan94** [comment]
> Custom CSS are worsening it , Current changes seems much better than that . Please Review

@kushthedude could you explain what you're trying to achieve ?

**Comment by kushthedude** [comment]
@niranjan94 I wanted to disable the internal scroll from the tax modal and for this i disabled the scrollable modal content , seems fine .
But when I use custom css to increase height , and when the checkbox for company info is ticked  it causes a body overflow thats why i didnt use custom css

**Comment by niranjan94** [comment]
> But when I use custom css to increase height , and when the checkbox for company info is ticked it causes a body overflow thats why i didnt use custom css

It overflows because you probably fixed the height. Instead you can use something like the `min-height` attribute to set a fixed height and also allow the div to increase in height as needed if more elements or added to the div.

**Comment by kushthedude** [comment]
@niranjan94 Please Review, No custom css needed tried fixing using the classes

**Comment by kushthedude** [comment]
@uds5501 @shreyanshdwivedi @mrsaicharan1 A peer review would be greate

**Comment by kushthedude** [comment]
@CosmicCoder96 @niranjan94 Review Please

**Comment by uds5501** [comment]
@kushthedude clicking send tax invoice option to true displaces the modal a little, why is that happening? Also, when I scroll down and start typing in, it scrolls up automatically. (produce able in deployement)

image

File: app/templates/components/forms/wizard/basic-details-step.hbs [hunk 2]
```diff
@@ -421,6 +421,11 @@
             </div>
             <div class="content">
               <div class="header">{{data.event.tax.name}}</div>
+              {{#if data.event.tax.shouldSendInvoice}}
+                <div class="description">
+                  {{t 'Company Name '}}{{data.event.tax.registeredCompany}}
```
Comment (uds5501) [inline app/templates/components/forms/wizard/basic-details-step.hbs]: A simple `{{data.event.tax.registeredCompany}}` would do. (I think we should follow the existing design of card and not introduce statics like 'Company Name'

Comment (kushthedude) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: @uds5501 That could be somewhat confusing than , As we will have `Tax Name` as well as `Company Name` , Thats why added a static

Comment (uds5501) [reply app/templates/components/forms/wizard/basic-details-step.hbs]: @kushthedude Okay, fair point. Please address the above issue then.

**Comment by kushthedude** [comment]
> @kushthedude clicking send tax invoice option to true displaces the modal a little, why is that happening? Also, when I scroll down and start typing in, it scrolls up automatically. (produce able in deployement)
> 
> image

@uds5501 will verify

**Comment by kushthedude** [comment]
@uds5501 Please have a look , The following problem was with the `UI Form` , Just made a quick tweak , See if its fine now

> @kushthedude clicking send tax invoice option to true displaces the modal a little, why is that happening? Also, when I scroll down and start typing in, it scrolls up automatically. (produce able in deployement)
> 
> image

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Please Review

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Please Review

File: app/templates/components/modals/tax-info-modal.hbs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div class="header">
   {{t 'Add tax information'}}
 </div>
-<div class="y-scrollable scrollable-modal content">
+<div class="{{if tax.shouldSendInvoice "y-scrollable scrollable-" ""}}modal content">
```
Comment (niranjan94) [inline app/templates/components/modals/tax-info-modal.hbs]: The extra `""` is not required

**Comment by kushthedude** [comment]
@niranjan94 @CosmicCoder96 Review now

**Comment by kushthedude** [comment]
@niranjan94 The random CI FAILURE again !

**Pull request merged**